Lib/getpass.py

```diff
@@ -68,12 +71,20 @@ def unix_getpass(prompt='Password: ', stream=None):
                 old = termios.tcgetattr(fd)     # a copy to save
                 new = old[:]
                 new[3] &= ~termios.ECHO  # 3 == 'lflags'
+                if echochar:
+                    new[3] &= ~termios.ICANON
                 tcsetattr_flags = termios.TCSAFLUSH
                 if hasattr(termios, 'TCSASOFT'):
                     tcsetattr_flags |= termios.TCSASOFT
                 try:
                     termios.tcsetattr(fd, tcsetattr_flags, new)
-                    passwd = _raw_input(prompt, stream, input=input)
+                    if not echochar:
+                        passwd = _raw_input(prompt, stream, input=input)
+                        stream.write('\n')
```

Comment:
Why the additioanal `stream.write('\n')`?

Reply:
Instead of a `return` you can do something like this:

```py
if echochar:
    passwd = _input_with_echochar(...)
else:
    passwd = _raw_input(...)
```

Reply:
you are right, it's just that initially there was a large structure there (before it was separated into a separate function) and I didn't want to move it further to the right
thanks for reminding me